vectorizers/_vectorizers.py

```diff
@@ -285,6 +285,46 @@ def preprocess_token_sequences(
     )
 
 
+def remove_node(adjacency_matrix, node, inplace=True):
+    if not inplace:
+        if scipy.sparse.isspmatrix_lil(adjacency_matrix):
+            adj = adjacency_matrix.copy()
+        else:
+            adj = adjacency_matrix.tolil()
+    elif not scipy.sparse.isspmatrix_lil(adjacency_matrix):
+        raise ValueError("Can only remove node in place from LIL matrices")
+    else:
+        adj = adjacency_matrix
+    # Copy the row we want to kill
+    row_to_remove = adj.rows[node].copy()
+    data_to_remove = adj.data[node].copy()
+    # Process all the rows making changes as required
+    for i in range(adj.rows.shape[0]):
+        if i == node:
+            adj.rows[i] = []
+            adj.data[i] = []
+        else:
+            try:
+                # Find out if this node has selected node as a successor
+                index_to_modify = adj.rows[i].index(node)
+                # If so replace the entry for that node with successor entries
+                # from the selected node
+                adj.rows[i][index_to_modify : index_to_modify + 1] = row_to_remove
```

Comment:
Nice trick to replace the one parent-child edge with the parent grand-children edges.